server/src/main/java/org/opensearch/rest/action/list/AbstractListAction.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.rest.action.list;
+
+import org.opensearch.client.node.NodeClient;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.rest.action.cat.AbstractCatAction;
+import org.opensearch.rest.pagination.PageParams;
+
+import java.io.IOException;
+import java.util.Objects;
+
+import static org.opensearch.rest.pagination.PageParams.PARAM_ASC_SORT_VALUE;
+import static org.opensearch.rest.pagination.PageParams.PARAM_DESC_SORT_VALUE;
+
+/**
+ * Base Transport action class for _list API.
```

Comment:
We can probably add additional documentation mentioning that this is for `_cat` api's that want to support pagination and give reference to indices api implementation.

Reply:
Added in the revision.